### PR TITLE
implement repositories attribute to gitea_team

### PR DESCRIPTION
**Pull request by yamadayutaka**
This PR implements the `repositories` attribute to `gitea_team` resource.

Details: #36

### CODE REVIEW

File: gitea/resource_gitea_team.go [hunk 1]
```diff
@@ -290,7 +308,75 @@ func resourceGiteaTeam() *schema.Resource {
...
+				}
+			}
+
+			page += 1
+		}
+	}
+
+	for repo, flag := range repositories {
+		if flag {
+			_, err = client.AddTeamRepository(team.ID, org, repo)
```
Comment (Lerentis) [inline gitea/resource_gitea_team.go:374]: wouldn't this do a add operation on each refresh in the worst case?
like in `resourceTeamRead` you set the update flag false but this code will still be executed, correcting a possible drift even the user did not yet apply anything. if my understanding is correct this needs to be fixed

**Review by Lerentis — CHANGES_REQUESTED** [review]
Thanks for raising this and directly working on it @yamadayutaka 😃 
i added an inline comment and apart from that could you please: 

* bump the version number by a minor release everywhere
* make the error handling in `setTeamRepositories` more explicit

Thanks

**Comment by yamadayutaka** [comment]
Hi @Lerentis , Thank you for your review!
I will check.

**Comment by Lerentis** [comment]
> > bump the version number by a minor release everywhere
> 
> Is it correct to bump the version from 0.12.3 to 0.13.0?

Yeah this should be fine as it is no breaking change

File: gitea/resource_gitea_team.go (hunk 1 above)
Comment (yamadayutaka) [reply gitea/resource_gitea_team.go:374]: The code is collapsed in the diff view, so I think you might be misunderstanding it.

`setTeamRepositories` is called from `resourceTeamCreate` and `resourceTeamUpdate`. (Not called from `resourceTeamRead`)

When calling from `resourceTeamCreate`, specify `false` for the `update` variable and perform the process of adding everything.

When called from `resourceTeamUpdate`, specify `true` for the `update` variable and perform add or remove operations depending on the result of matching with existing repositories.

**Comment by yamadayutaka** [comment]
>make the error handling in setTeamRepositories more explicit

Could you elaborate a bit more on your concerns?

**Comment by Lerentis** [comment]
> > make the error handling in setTeamRepositories more explicit
> 
> Could you elaborate a bit more on your concerns?

currently you just return on error != nil. it would be good to get a proper error message

**Comment by yamadayutaka** [comment]
I have done the following:

- Bump the version from 0.12.3 to 0.13.0
- Print log message on error in `setTeamRepositories`

**Comment by Lerentis** [comment]
maybe i was not clear enough, so code is better than english : https://github.com/Lerentis/terraform-provider-gitea/commit/19062e27f0b57bdb56d4fe716af763dec3501c04

File: gitea/resource_gitea_team.go (hunk 1 above)
Comment (Lerentis) [reply gitea/resource_gitea_team.go:375]: indeed i read that wrong in the diff view. need to get back to the habit of checking everything locally. thanks for the clarification

**Comment by yamadayutaka** [comment]
Thanks! I synced you commit.

**Review by Lerentis — APPROVED** [review]
LGTM
Thanks for the implementation and the prompt responses 😃

**Pull request merged**